### ISSUE TITLE
Usage of curly braces in ARM assembly considered invalid

### ISSUE THREAD

**Issue by mahircg-aws** [rust-lang/rust#140963]
ARM multi load/store instructions require curly braces to specify the target register range: https://developer.arm.com/documentation/ddi0406/c/Application-Level-Architecture/The-Instruction-Sets/Load-store-multiple-instructions?lang=en#:~:text=Load%20Multiple%20instructions%20load%20a,are%20consecutive%20word%2Daligned%20words.

The following inline assembly instruction requires curly braces as part of specifying the register range.

```rust
unsafe { asm!("stmib sp, {x0 - x15}",) }
```

During compilation, I get the following error. 

```
error: invalid asm template string: expected `}`, found `-`
   --> firecrab/src/main.rs:386:34
    |
386 |     unsafe { asm!("stmib sp, {x0 - x15}",) }
    |                              -   ^ expected `}` in asm template string
    |                              |
    |                              because of this opening brace
    |
    = note: if you intended to print `{`, you can escape it using `{{`
```

Reading https://rust-lang.github.io/rfcs/2873-inline-asm.html#use-of-double-braces-in-the-template-string, it looks like the only usage for curly braces was for AVX extension in x86. 

Same error is reported on assembly that is included in Rust world with `core::arch::global_asm!(include_str!("asm.s"));`

```
.macro asm_test
    stmib sp, {x0 - x15}
```

Is that a bug in the way the assembly blobs are parsed?

### Meta

`rustc --version --verbose`:
```
rustc 1.84.0-nightly (b19329a37 2024-11-21)
binary: rustc
commit-hash: b19329a37cedf2027517ae22c87cf201f93d776e
commit-date: 2024-11-21
host: aarch64-apple-darwin
release: 1.84.0-nightly
LLVM version: 19.1.4
```

**Comment by taiki-e** [comment]
You have to use `{{x0 - x15}}` (double `{` and `}`) or [`raw` option](https://doc.rust-lang.org/nightly/reference/inline-assembly.html#r-asm.options.supported-options.raw).

**Comment by mahircg-aws** [comment]
I tried that, and it causes the following error. I think it interprets the instruction as a vector operation and expects vector registers.

```
error: <instantiation>:2:16: vector register expected
    stmib sp, {x0 - x15}
               ^

error: could not compile `bin` (bin "bin")
```
